back-end/src/sciler/config/configHandler.go

```diff
@@ -39,160 +38,254 @@ func generateDataStructures(readConfig ReadConfig) (WorkingConfig, error) {
 		}
-	case "string-array":
-		if reflect.TypeOf(value).Kind() != reflect.Slice {
-			return errors.New("Value was not of type array: " + fmt.Sprint(value))
+	} else if logic["type"] != nil && reflect.TypeOf(logic["type"]).Kind() == reflect.String && logic["type_id"] != nil && reflect.TypeOf(logic["type_id"]).Kind() == reflect.String {
+		condition := Condition{
+			Type:        logic["type"].(string),
+			TypeID:      logic["type_id"].(string),
+			Constraints: generateLogicalConstraint(logic["constraints"]),
 		}
-		array := value.([]interface{})
-		for i := range array {
-			if reflect.TypeOf(array[i]).Kind() != reflect.String {
-				return errors.New("Value was not of type string-array: " + fmt.Sprint(value))
+
+		return condition
+	}
+	panic(fmt.Sprintf("JSON config in wrong condition format, conditions: %v, could not be processed", conditions))
+}
+
+func generateLogicalConstraint(constraints interface{}) LogicalConstraint {
+	logic := constraints.(map[string]interface{})
+	if logic["operator"] != nil { // operator
+		if logic["operator"] == "AND" {
+			and := AndConstraint{}
+			for _, constraint := range logic["list"].([]interface{}) {
+				and.logics = append(and.logics, generateLogicalConstraint(constraint))
 			}
+			return and
+		} else if logic["operator"] == "OR" {
+			or := OrConstraint{}
+			for _, constraint := range logic["list"].([]interface{}) {
+				or.logics = append(or.logics, generateLogicalConstraint(constraint))
+			}
+			return or
+		} else {
+			panic(fmt.Sprintf("JSON config in wrong format, operator: %v, could not be processed", logic["operator"]))
 		}
-	case "bool-array":
-		if reflect.TypeOf(value).Kind() != reflect.Slice {
-			return errors.New("Value was not of type array: " + fmt.Sprint(value))
-		}
-		array := value.([]interface{})
-		for i := range array {
-			if reflect.TypeOf(array[i]).Kind() != reflect.Bool {
-				return errors.New("Value was not of type bool-array: " + fmt.Sprint(value))
+	} else if logic["comp"] != nil && reflect.TypeOf(logic["comp"]).Kind() == reflect.String {
+		var constraint Constraint
+		if logic["component_id"] != nil && reflect.TypeOf(logic["component_id"]).Kind() == reflect.String {
+			constraint = Constraint{
+				Comparison:  logic["comp"].(string),
+				ComponentID: logic["component_id"].(string),
+				Value:       logic["value"],
+			}
+		} else if logic["component_id"] == nil {
+			constraint = Constraint{
+				Comparison:  logic["comp"].(string),
+				ComponentID: "",
+				Value:       logic["value"],
 			}
+		} else {
+			panic(fmt.Sprintf("JSON config in wrong format, component_id should be of type string, %v is of type %s", logic["component_id"], reflect.TypeOf(logic["component_id"]).Kind().String()))
 		}
-	default:
-		// Value is of custom type, must be checked at client computer level
-		return nil
+
+		return constraint
 	}
-	return nil
+	panic(fmt.Sprintf("JSON config in wrong constraint format, conditions: %v, could not be processed", constraints))
 }
 
```

Comment:
Can you explain why this code is commented or still here

Reply:
Done